homework-submission/week2/my-app/src/components/ToDoItems.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react"
+import Post from "../Post"
+
+import ToDos from "../Todos"
+
+class TodoItem extends Component {
+   constructor(props) {
+      super(props)
+      this.state = {
+         Todos: ToDos
+      }
+      this.onChange = this.onChange.bind(this)
+   }
+
+   onChange(id) {
+      this.setState(prevState => {
+         const updatedTodos = prevState.Todos.map(todo => {
+            if (todo.id === id) {
+               todo.completed = !todo.completed
+            }
+            return todo
+         })
+         return {
+            Todos: updatedTodos
+         }
+      })
+   }
+
+   render() {
+      const blank = <h2>No items...</h2>;
+      return (
+         this.state.Todos.length === 0 ? blank :
+            <div>
+               {
+                  ToDos.map((item, id) => (
+                     <Post
+                        checkbox={<input type="checkbox" checked={item.completed} onChange={() => this.onChange(item.id)} />}
```

Comment:
you are passing the HTML element as prop and that is not good.
you have you create this element in the Post component.